### PR TITLE
Add create node with options

### PR DESCRIPTION
**Pull request by pablogs9**
This PR adds a node initializer with options in order to for example be able to modify the Domain ID from the RCLC

### CODE REVIEW

**Review by jamoralp — APPROVED** [review]

**Comment by JanStaschulat** [comment]
@pablogs9 @jamoralp looks ok. Please also add a unit test in https://github.com/micro-ROS/rclc/blob/master/rclc/test/rclc/test_node.cpp and run 'colcon test'

File: rclc/include/rclc/node.h [hunk 1]
```diff
@@ -49,6 +49,33 @@ rclc_node_init_default(
...
+ *
+ * \param[in] name the name of the node
+ * \param[in] namespace the namespace of the node
+ * \param[in] support the rclc_support_t object
+ * \param[in] node_ops node options
+ * \return rcl_node_t if successful
+ * \return NULL if an error occurred
+ */
+rcl_ret_t
+rclc_node_init_with_options(
```
Comment (JanStaschulat) [inline rclc/include/rclc/node.h:71]: Please add for every new function also a unit test in the corresponding unit test file.

Comment (pablogs9) [reply rclc/include/rclc/node.h:71]: Working on it, @jamoralp is preparing a PR with these tests. Sorry for that Jan, we were in a hurry for these options and I forgot to implement the tests.

Comment (jamoralp) [reply rclc/include/rclc/node.h:71]: Done. Please check #46 . Sorry for the inconvenience.

**Pull request merged**